### ISSUE TITLE
Bug: cannot find "@typescript-eslint/types" when typescript version lower than 3.8

### ISSUE THREAD

**Issue by Darkylin** [typescript-eslint/typescript-eslint#5332]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### Repro Code

```typescript
packages/types/package.json: line 13
  "files": [  // <=== lost "_ts3.4"
    "dist",
    "package.json",
    "README.md",
    "LICENSE"
  ],
```
``` packages/types/package.json: line 72
 "typesVersions": {
    "<3.8": {
      "*": [
        "_ts3.4/*"
      ]
    }
  },
```
```


### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

could find "@typescript-eslint/types" when typescript version lower than 3.8

### Actual Result

cannot find "@typescript-eslint/types" when typescript version lower than 3.8

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `X.Y.Z` |
| `@typescript-eslint/parser`            | `X.Y.Z` |
| `@typescript-eslint/scope-manager`     | `X.Y.Z` |
| `@typescript-eslint/typescript-estree` | `X.Y.Z` |
| `@typescript-eslint/type-utils`        | `X.Y.Z` |
| `@typescript-eslint/utils`             | `5.30.5` |
| `TypeScript`                           | `3.7` |
| `ESLint`                               | `X.Y.Z` |
| `node`                                 | `X.Y.Z` |

**Comment by bradzacher** [comment]
Out of curiosity why are you using a TS version that old? TS 3.8 itself is almost 3 years old at this point.

It seems strange to be building new rules with our tooling using such an out-of-date version of TS.

**Comment by Darkylin** [comment]
Well, it's a legacy monolith project, there are many teams maintaining together. Too difficult to update😂
